test/git_utils.py

```diff
@@ -0,0 +1,45 @@
+from subprocess import check_output
+
+
+def clone(upstream_repo_path, destination):
+    return check_output(
+        [
+            "git",
+            "clone",
+            # Avoids installing default hooks
+            # the user might have configured
+            "--template",
+            "/dev/null",
+            upstream_repo_path,
+            destination,
+        ]
+    )
+
+
+def init(repo_path, user="Test User", email="testuser@example.com"):
+    run(repo_path, "-c", "init.defaultBranch=master", "init", "--template", "/dev/null")
+    run(repo_path, "config", "--local", "user.name", user)
+    run(repo_path, "config", "--local", "user.email", email)
+
+
+def commit_all(repo_path, msg=None):
+    if msg is None:
+        msg = "Committing all changes"
+
+    run(repo_path, "add", ".")
+    run(repo_path, "commit", "-m", msg)
```

Comment:
Does this work if the user has not globally configured a git user name and e-mail?

Reply:
No, and the testsuite fails without them, I saw that issue but wrongfully thought setting the user in `init` was enough. Clearly it isn't when calling `commit_all` on a repository that was initialized externally.

Reply:
```
COMMIT INFORMATION
       Author and committer information is taken from the following environment variables, if set:

           GIT_AUTHOR_NAME
           GIT_AUTHOR_EMAIL
           GIT_AUTHOR_DATE
           GIT_COMMITTER_NAME
           GIT_COMMITTER_EMAIL
           GIT_COMMITTER_DATE
```
From `man git-commit`.